frontend/src/app/offers/offer-detail/offer-detail.component.ts

```diff
@@ -24,6 +25,11 @@ export class OfferDetailComponent implements OnInit {
 
   ngOnInit() {
     this.offer$ = this.activatedRoute.params
-    .switchMap(params => this.offersService.getOffer(params.offerId));
+    .switchMap(params => this.offersService.getOffer(params.offerId))
+    .map(offer => {
+      offer.joined = false;
```

Comment:
@jacol12345 this is frontend part of #416 - there's also backend to be developed (see #920).

I believe @OtisRed added this `offer.joined = false` just to fulfill `Offer` type interface.

@OtisRed most probably TypeScript is not able to check if offer, that will come from backend contains this attribute, so there's no reason to add this `map` part here (type checking phase is run during transpilation from TypeScript to JavaScript, no when code is executed in browser). Can You try, if it will wokr without this `map` part?

Reply:
@jacol12345 it was set to false, because when set to "true" ngIf directive in .html file was jumping to the ng-template pointed by "else". I didn't know what to do with that.  

@magul Not sure if I get it. As I understood this earlier, the .map method was calling the offer to check if the offer.model.ts has the property "joined" and then read that it's boolean, so that true/false default (or any real value in the future) could be read by *ngIf directive in .html file. If it's right, it made sense to me. Without the .map method I'm kinda lost though. I deleted the map part so it looked like this:
    offer => {
      offer.joined = true;
      return offer;
and it did not work. Changing the value for true or false causes no difference in ngIf directive. But I'm on the way too early stage of tutorial to understand what I'm doing here right now :P

Reply:
@OtisRed if You will not set it explicitly to `false` JavaScript will evaluate it to `undefined`, that will work the same way as `false`

You can leave it like:
```
this.offer$ = this.activatedRoute.params
  .switchMap(params => this.offersService.getOffer(params.offerId));
```
this `map` that adds `joined = false` is not required, as TypeScript will not know that offer, that comes from backend doesn't have this attribute set to boolean value (and JavaScript will evaluate it to `undefined`)

Reply:
IMO Offer need a proper model class and the constructor to force the properties initialization (inside the service), the same as we talked briefly about User. However, we can leave it as is for later redesign I guess.